app/serializers/api/v2/measures/measure_serializer.rb

```diff
@@ -8,8 +8,23 @@ class MeasureSerializer
 
         set_id :measure_sid
 
-        attributes :id, :origin, :effective_start_date, :effective_end_date, :import,
-                   :excise, :vat, :reduction_indicator
+        attributes :id,
+                   :origin,
+                   :effective_start_date,
+                   :effective_end_date,
+                   :import,
+                   :excise,
+                   :vat,
+                   :reduction_indicator,
+                   :meursing
+
+        attribute :resolved_duty_expression do |root_measure, params|
+          additional_code_id = params[:meursing_additional_code_id]
+
+          if additional_code_id.present?
+            root_measure.resolved_duty_expression_for(additional_code_id)
```

Comment:
If I'm reading this correctly - this introduces an N+1

The MeasureSerializer will be run _once per measure_ 
- if there is an `additional_code_id.present?` 
- call `resolved_duty_expression_for`
- which calls `meursing_measures_for`
- if its a meursing measure, it then calls `MeursingMeasureFinderService`
- which in then performs a DB query?

Reply:
So it does but I'm not so worried about it for a couple of reasons:

1. It performs fine in real life testing for even really heavy measures
2. We only do this for a seriously small subset of commodity requests
3. The number of measures isn't set to grow to beyond the number of measures I've tested against in the worst case for the subset of measures I've dealt with
4. We have indexes for most of the filters we're using from my understanding

If we do see a performance hit because some of these changes we can eager load a sefl-referential join as part of the cached commodity service process and extend the list of indexes. I suspect we'll never need to do this, though